components/new_channel_modal/new_channel_modal.tsx

```diff
@@ -108,12 +97,11 @@ const NewChannelModal = () => {
     const [serverError, setServerError] = useState('');
 
     // create a board along with the channel
+    const pluginsComponentsList = useSelector((state: GlobalState) => state.plugins.components);
+    const createBoardFromChannelPlugin = pluginsComponentsList?.CreateBoardFromTemplate;
     const [addBoard, setAddBoard] = useState(false);
-    const [selectedBoardTemplate, setSelectedBoardTemplate] = useState<BoardTemplate | null>(null);
-    const [boardTemplates, setBoardTemplates] = useState<BoardTemplate[]>([]);
     const newChannelWithBoardPulsatingDotState = useSelector((state: GlobalState) => getPreference(state, Preferences.APP_BAR, Preferences.NEW_CHANNEL_WITH_BOARD_TOUR_SHOWED, ''));
-    const EMPTY_BOARD = 'empty-board';
-    const focalboardEnabled = useSelector(canCreateBoards);
+    const [selectedBoardTemplateId, setSelectedBoardTemplateId] = useState<string | null>(null);
```

Comment:
@hmhealey thanks for the feedback and the example code. I liked your approach and implemented it since I agree is cleaner and moves more responsability to the plugin 👍